cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/sourcecontrol/SourceControlHelper.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.sourcecontrol;
+
+import io.cdap.cdap.proto.id.ApplicationId;
+import io.cdap.cdap.sourcecontrol.SourceControlException;
+import io.cdap.cdap.sourcecontrol.operationrunner.PullAppResponse;
+import java.util.Collection;
+
+/**
+ * Provides various helper methods for source control operations. Would be implemented for both
+ * running in app-fabric and running in workers.
+ */
+public interface SourceControlHelper {
+
+  /**
+   * Deploys a given app version with the given pull app details.
+   *
+   * @param appId the {@link ApplicationId} for the app version to be deployed
+   * @param pullDetails {@link PullAppResponse} which includes the app spec and the git hash.
+   */
+  void deployAppVersion(ApplicationId appId, PullAppResponse<?> pullDetails)
+      throws SourceControlException;
+
+  /**
+   * Mark the given list of app-versions as the latest. Only the latest version for any app is
+   * runnable.
+   *
+   * @param appIds List of {@link ApplicationId} to be marked latest
+   */
+  void markAppVersionsLatest(Collection<ApplicationId> appIds) throws SourceControlException;
+
+  /**
+   * Delete the given app versions.
```

Comment:
is this deleting all versions of the app? Don't think we would want to do that.

If it's deleting a single version, don't think we have any way to do that, and don't think we really need to either.

Reply:
Removed the deletion step